OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -47,12 +47,12 @@ class CellTrigger
 			public readonly CPos[] Footprint;
 			public bool Dirty;
 
-			readonly Action<Actor> onActorEntered;
-			readonly Action<Actor> onActorExited;
+			readonly Action<Tuple<Actor, Player>> onActorEntered;
+			readonly Action<Tuple<Actor, Player>> onActorExited;
```

Comment:
I think you missed an important detail here: if the unit leaves the trigger area then *you discard/ignore it from the owner changed list*.  `OnExited` should never be able to run twice for the same actor in the same tick.